zulipterminal/model.py

```diff
@@ -729,17 +730,55 @@ def toggle_stream_pinned_status(self, stream_id: int) -> bool:
     def is_user_subscribed_to_stream(self, stream_id: int) -> bool:
         return stream_id in self.stream_dict
 
+    def make_reduced_stream_data(self, stream: Any) -> StreamData:
+        # stream_id has been changed to id.
+        return StreamData({'name': stream['name'],
+                           'id': stream['stream_id'],
+                           'color': stream['color'],
+                           'invite_only': stream['invite_only'],
+                           'description': stream['description']})
+
+    def get_stream_by_id(self, streams: List[StreamData], stream_id: int
+                         ) -> StreamData:
+        for stream in streams:
+            if stream['id'] == stream_id:
+                return stream
+        raise RuntimeError("Invalid stream id.")
+
+    def update_sidebar(self) -> None:
+        self.controller.view.left_panel.update_stream_view()
+        self.controller.update_screen()
+
+    def add_stream(self, stream_info: Any) -> None:
+        stream_data = self.make_reduced_stream_data(stream_info)
+        self.stream_dict[stream_info['stream_id']] = stream_info
```

Comment:
We can't use an assert over here, considering the scenarios when the user has been unsubscribed and then subscribed to a stream from web UI. This would cause the assert to fail.

Reply:
I agree, based on the point you made in the stream.